mypy/semanal.py

```diff
@@ -6408,14 +6408,44 @@ def lookup_fully_qualified_or_none(self, fullname: str) -> SymbolTableNode | Non
         # TODO: support nested classes (but consider performance impact,
         #       we might keep the module level only lookup for thing like 'builtins.int').
         assert "." in fullname
-        module, name = fullname.rsplit(".", maxsplit=1)
-        if module not in self.modules:
```

Comment:
Please keep the original "fast path" which we'll be hitting most of the time. This way your new logic, which is a bit more expensive, it not slowing he typical code path. This may be performance critical. In more concrete terms, add your new logic under `if module not in self.modules:`, and keep the other code as is in the function.